torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -413,22 +419,17 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
                 clientPipe.pendingResponseMessage_.erase(it);
               }
 
-              threadPool_.run(
-                  [this,
-                   futureResponseMessage,
-                   responseMessage{std::move(responseMessage)}]() mutable {
-                    if (responseMessage.type() == MessageType::EXCEPTION) {
-                      markFutureWithError(
-                          std::move(futureResponseMessage),
-                          std::string(
-                              responseMessage.payload().begin(),
-                              responseMessage.payload().end()));
-                    } else {
-                      markFutureAsComplete(
-                          std::move(futureResponseMessage),
-                          std::move(responseMessage));
-                    }
-                  });
+              if (responseMessage.type() == MessageType::EXCEPTION) {
+                markFutureWithError(
+                    std::move(futureResponseMessage),
+                    std::string(
```

Comment:
Other part of the code bases uses `c10::str`/`c10::toString`.

Reply:
`c10::toString` gave me an error: my impression is that it's only useful to convert some common types to a meaningful string representation. `c10::str` didn't give me an error but I don't think it does what we want: from the code it seems it just appends (`<<`) each argument, which in our case would print the start and end iterator, whereas what we want is copy the string. Did I miss something?

Note that the ProcessGroup agent is also using `std::string` for this:
https://github.com/pytorch/pytorch/blob/b636f5e324f19ebed867d6c3088580a6d6793859/torch/csrc/distributed/rpc/process_group_agent.cpp#L591-L592

Reply:
I see, thanks for explaining.